internal/cli/importcmd/yaml.go

```diff
@@ -43,44 +44,77 @@ type TenantConfig struct {
 	} `yaml:"roles"`
 }
 
-func (t *TenantConfig) getYAML(yamlData string, config string) *TenantConfig {
-	type StructA struct {
-		A string `yaml:"a"`
+func ParseYAML(yamlPath string, config *Config) *TenantConfig {
+
+	yamlData, err := ioutil.ReadFile(yamlPath)
+	if err != nil {
+		log.Printf("error reading yaml #%v ", err)
 	}
-	var a StructA
 
-	err := yaml.Unmarshal([]byte(yamlData), &a)
+	t := &TenantConfig{}
+
+	err = yaml.Unmarshal(yamlData, t)
 	if err != nil {
-		log.Fatalf("cannot unmarshal data: %v", err)
+		log.Fatalf("Unmarshal yaml file: %v ", err)
 	}
 
-	// Unmarshall yaml
+	for key, replacement := range config.Auth0KeywordReplaceMappings {
+		t.replaceClientConfig(key, replacement)
+		t.replaceResourceServersConfig(key, replacement)
+		t.replaceRolesConfig(key, replacement)
+	}
 
-	// yamlFile, err := ioutil.ReadFile(yaml)
-	// if err != nil {
-	// 	log.Printf("error reading yaml #%v ", err)
-	// }
+	return t
+}
 
-	// // err = yaml.Unmarshal(yamlFile, t)
-	// // if err != nil {
-	// // 	log.Fatalf("Unmarshal yaml file: %v", err)
-	// // }
+func (t *TenantConfig) replaceClientConfig(key string, replacement interface{}) {
+	for i, client := range t.Clients {
 
-	// Unmarshall config json
+		if strings.ContainsAny(client.Name, key) {
+			str := strings.ReplaceAll(client.Name, fmt.Sprintf("##%s##", key), replacement.(string))
+			t.Clients[i].Name = str
 
-	configFile, err := ioutil.ReadFile(config)
-	if err != nil {
-		log.Printf("error reading config json #%v ", err)
-	}
+		}
 
-	c := &Config{}
+		for j, url := range client.AllowedLogoutUrls {
```

Comment:
Lots of repetition here. Will clean up, but at least the interface is there to work with.

Reply:
Don't worry about that.